packages/providers/file-system/src/Client/FileSystemManager.ts

```diff
@@ -0,0 +1,217 @@
+/**
+ * Copyright 2024 Mytra Control S.L. All rights reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be found in the LICENSE file
+ * or at https://opensource.org/licenses/MIT.
+ */
+
+import { Crash, Multi } from '@mdf.js/crash';
+import { DebugLogger, LoggerInstance, SetContext } from '@mdf.js/logger';
+import fs from 'fs';
+import _ from 'lodash';
+import { EventEmitter } from 'stream';
+import { v4 } from 'uuid';
+import {
+  CopyOptions,
+  FileSystemManagerOptions,
+  ReadDirOptions,
+  ReadOptions,
+  WriteOptions,
+} from './types';
+
+export declare interface FileSystemManager {
```

Comment:
-  the client should manage the file, not give an interface to manage the file externaly, this means that the class should offer the public function 'append', the rest is part of the logic of this class. 
- this logic should rotate the file, move the file to other folder, move old files to other folder ...

Reply:
Done.